crates/rolldown/src/utils/chunk/finalize_chunks.rs

```diff
@@ -56,10 +59,9 @@ pub fn finalize_assets(
     index_vec![Xxh3::default(); preliminary_assets.len()];
 
   let index_final_hashes: IndexVec<AssetIdx, String> = index_asset_hashers
-    .into_iter_enumerated()
-    // FIXME: Extra traversing. This is a workaround due to `par_bridge` doesn't ensure order https://github.com/rayon-rs/rayon/issues/551#issuecomment-882069261
-    .collect::<Vec<_>>()
+    .raw
     .into_par_iter()
+    .enumerate()
     .map(|(asset_idx, mut hasher)| {
```

Comment:
`asset_idx` become `usize` instead of `AssetIdx`, this is unexpected. I'm not sure why `oxc_index` allows using `usize` as the valid index type. But the reason we use `IndexVec` is to have a domain-specific `Vec` to prevent misusing wrong index. Otherwise we could just use std `Vec`.

Reply:
@overlookmotel Sorry for bothering. Could you take have a rough look if this is expected/reasonable behavior of `oxc_index`?

Reply:
@hyf0 Sorry I'm so late to come back on this. It got lost in my Github notifications.

I've made a PR to prevent `IndexVec`s being indexed into with a `usize` (https://github.com/oxc-project/oxc/pull/5733). As you say, that's a bad idea! That oddity was inherited from `index_vec` crate.

Opened an issue for the par iter problem: https://github.com/oxc-project/oxc/issues/5734